tests/test_llm_github_models.py

```diff
@@ -212,3 +212,60 @@ async def test_async_model_prompt():
     model = get_async_model("github/gpt-4.1-mini")
     response = await model.prompt("What is the capital of France?")
     assert "Paris" in await response.text()
+
+
+def test_sync_returns_usage():
+    """
+    Test that the sync model returns usage information for streaming and non-streaming.
+    """
+    model = get_model("github/gpt-4.1-mini")
+    response = model.prompt("What is the capital of France?")
+    usage = response.usage()
+
+    assert usage is not None
+    assert usage.input > 0
+    assert usage.output > 0
+
+    response = model.prompt("What is the capital of France?", stream=True)
+    usage = response.usage()
+
+    assert usage is not None
+    assert usage.input > 0
+    assert usage.output > 0
+
+
+def test_usage_returns_details():
+    """
+    Test that models which return extra usage details (e.g. thinking models) return them correctly.
+    """
+    model = get_model("github/o3-mini")
```

Comment:
@tonybaloney I needed to use a reasoning model here to get the extra output, not sure if this is going to be OK with rate limiting - let me know

Reply:
One of the Phi-4 models supports reasoning and is the `low` tier of rate-limits. o3-mini is going to cause problems because we only get a [couple of requests a day](https://docs.github.com/en/github-models/use-github-models/prototyping-with-ai-models#rate-limits).

Reply:
Yup makes sense - I reworked the test to just check `set_usage` directly rather than going through a real prompt